File: servicebus-master/src/main/java/net/trajano/servicebus/master/MasterActor.java
package net.trajano.servicebus.master;

import akka.actor.UntypedActor;

public class MasterActor extends UntypedActor {
	// providers.foreach(provider => provider.handles(classOf[_]) &&
	// provider.newActor(context) ! _)
	//
	// private final Set<ActorProvider> providers = new HashSet<>(); def receive
	// = {
	// case ActorRegistration(provider) => {
	// log.info("received test")
	// System.out.println("!!!!")
	// providers += provider
	// }
	// case _ => {
	//
	// }
}

File: servicebus-master/src/main/scala/net/trajano/servicebus/master/MasterActor.scala
package net.trajano.servicebus.master

import akka.actor.Actor
import akka.actor.ActorLogging
import scala.collection.mutable.Set

class MasterActor extends Actor with ActorLogging {
  val providers = Set[ActorProvider]()
  def receive = {
    case ActorRegistration(provider) => {
      log.info("received test")
      System.out.println("!!!!")
      providers += provider
    }
    case _ => {
      val message = _
      // providers.foreach(provider => provider.handles(classOf[_]) && provider.newActor(context) ! _)
    }
  }
}